client/me/concierge/pending-appointment/pending-appointment.js

```diff
@@ -0,0 +1,98 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import getConciergeNextAppointment from 'state/selectors/get-concierge-next-appointment';
+import PrimaryHeader from '../shared/primary-header';
+import Site from 'blocks/site';
+import { localize, moment } from 'i18n-calypso';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+class PendingAppointment extends Component {
+	static propTypes = {
+		nextAppointment: PropTypes.object.isRequired,
+		site: PropTypes.object.isRequired,
+	};
+
+	componentDidMount() {
+		this.props.recordTracksEvent( 'calypso_concierge_pending_appointment' );
+	}
+
+	render() {
+		const { nextAppointment, site, translate } = this.props;
+
+		const appointmentDateTime = moment( nextAppointment.beginTimestamp )
+			.tz( nextAppointment.meta.timezone )
+			.format( 'LLLL z' );
+
+		const cancelLink = '/me/concierge/' + site.slug + '/' + nextAppointment.id + '/cancel';
```

Comment:
And on second thought — we're re-creating these links an awful lot. It may be useful to abstract these into helper functions in something like `client/me/concierge/utils.js` to ensure consistency in case the URL ever changes.